server/webhook_parser_misc_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/andygrunwald/go-jira"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMarkdown(t *testing.T) {
+	f, err := os.Open("testdata/webhook-issue-created.json")
+	require.NoError(t, err)
+	defer f.Close()
+	wh, _, err := ParseWebhook(f)
+	require.NoError(t, err)
+	w := wh.(*webhook)
+	require.NotNil(t, w)
+	require.Equal(t,
+		"Test User created story [TES-41](https://some-instance-test.atlassian.net/browse/TES-41)",
+		w.headline)
+}
+
+func TestWebhookVariousErrors(t *testing.T) {
+	assert.Equal(t, "", mdUser(nil))
+
+	wh := &webhook{
+		JiraWebhook: &JiraWebhook{
+			Issue: jira.Issue{
+				Fields: &jira.IssueFields{},
+			},
+		},
+	}
+
+	assert.Equal(t, "", wh.mdJiraLink("test", "/test"))
+	assert.Equal(t, "", wh.mdIssueDescription())
+	assert.Equal(t, "", wh.mdIssueSummary())
+	assert.Equal(t, "_nobody_", wh.mdIssueAssignee())
+	assert.Equal(t, "", wh.mdIssueType())
+	assert.Equal(t, " ", wh.mdSummaryLink())
```

Comment:
A single space?

Reply:
These tests were originally added for coverage and/or check for nil pointers and default handling. 1/5 These functions do return an `str1 + " " + str2` and are not supposed to be called with nil/defaults.

Reply:
Ya that's fine then.